front/src/applications/operationalStudies/hooks.ts

```diff
@@ -25,13 +26,13 @@ import type { ManageTrainSchedulePathProperties } from './types';
 /**
  * Hook to relaunch the pathfinding when editing a train
  */
+
 export const useSetupItineraryForTrainUpdate = (
-  setPathProperties: (pathProperties: ManageTrainSchedulePathProperties) => void
+  setPathProperties: (pathProperties: ManageTrainSchedulePathProperties) => void,
+  trainIdToEdit: number | undefined
```

Comment:
There is a typescript error on l.51 because of this. You can remove the undefined because we call this hook only if trainIdToEdit is defined.

```suggestion
  trainIdToEdit: number
```